umeboshi/serializer.py

```diff
@@ -0,0 +1,37 @@
+import pickle
+from importlib import import_module
+from django.core.exceptions import ImproperlyConfigured
+from django.conf import settings
+
+class DefaultSerializer(object):
+    def dumps(self, value):
+        return pickle.dumps(value)
+
+    def loads(self, value):
+        return pickle.loads(value)
+
+
+def load_class(path):
+    """
+    Loads class from path.
+    """
+
+    mod_name, klass_name = path.rsplit('.', 1)
+
+    try:
+        mod = import_module(mod_name)
+    except AttributeError as e:
+        raise ImproperlyConfigured('Error importing {0}: "{1}"'.format(mod_name, e))
+
+    try:
+        klass = getattr(mod, klass_name)
+    except AttributeError:
+        raise ImproperlyConfigured('Module "{0}" does not define a "{1}" class'.format(mod_name, klass_name))
+
+    return klass
```

Comment:
If we are going to return this `klass`, we should validate that it's `type`. Like, it should implement a certain base class or interface, right?

Reply:
I wasn't sure if we want to make a base class rather then `object` for user to inherent. An interface makes sense, here are two options I can think of:

**1. Validate class method to check class contains `dumps` and `loads`  method**:
Pros: Developer can implement custom serializer out of object and reuse it in different places. ex. `Django-Redis` and `Umeboshi` (see this [PR](https://github.com/makingspace/site/pull/11344))
Cons: I can't think of any

**2. Validate class type to match a `umeboshi` base serializer class:**
Pros: Identify error quicker during app initialization time
Cons: Custom serializer needs to inherent from `umeboshi` base class which might not be reusable for other places  

@joeymink Any feedback?

Reply:
I dig your option 1!

Reply:
Done

Reply:
I am in favor of the base class/inheritance option(2).

Reply:
I didn't understand the reusability con.

Reply:
In this [PR](https://github.com/makingspace/site/pull/11344), if we go with option 2 then we need to make one serializer for `Django-Redis`, and another one for `Umeboshi`, with exactly same methods.

Reply:
What is wrong with Umeboshi having both(Py2 and Py3) serializers, and the client decides which one to use?

Reply:
We can achieve the same goal for Py2 Py3 migration is we make options granular enough like the following:

```
UMEBOSHI_SERIALIZER = {
    DUMPS: {
        PICKLE_PROTOCOL: Number
    },
    LOADS: {
        ENCODING: String
    }
}
```

Then in our Site config, we can generate options based on Py2 or Py3:
```
UMEBOSHI_SERIALIZER = {
    DUMPS: {
        PICKLE_PROTOCOL: 2
    },
    LOADS: {
        ENCODING: "latin1" if PY3 else "default"
    }
}
```

Perhaps this looks preferable to you? I think both works. In `Django-Redis`, `PICKLE_PROTOCOL` is an explicit config option ([link](https://django-redis-cache.readthedocs.io/en/latest/advanced_configuration.html#pickle-version))

More over, we can support both custom serializer & granular config options